src/main/java/com/klarna/hiverunner/HiveServerContainer.java

```diff
@@ -117,6 +117,10 @@ public Path getBaseDir() {
 
     public List<Object[]> executeStatement(String hiveql) {
         try {
+            if(isViewJoin(hiveql)) {
+              int index = hiveql.toLowerCase().indexOf(" on ");
+              hiveql = hiveql.substring(0, hiveql.toLowerCase().indexOf(" on ")) + hiveql.substring(index, hiveql.length()).toLowerCase() ;
```

Comment:
So is the trick here that we need to lower case the word "on" and also the table name that follows? Is there no way that anything other than table name could be included in the rest of the string? Can a view join be used in a sub-select or other things? If so there could be problems with lower-casing everything else. This feels really brittle but having only the SQL string available as opposed to the parsed tree makes it painful.